### PR TITLE
LPS-186637 | Allow creating POST method on API Builder

### PR DESCRIPTION
**Pull request by jaimelr10**
**Jira Ticket**: https://liferay.atlassian.net/browse/LPS-186637

**Description**: Allow creating and editing POST methods on API Builder with its required fields in info and configuration tabs

**Screenshots**:
image
image
image
image

### CODE REVIEW

File: modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx [hunk 1]
```diff
@@ -180,82 +248,105 @@ export default function BaseAPIEndpointFields({
...
-
 			<ClayForm.Group
 				className={classNames({
-					'has-error': displayError.retrieveType,
+					'has-error': displayError.httpMethod,
 				})}
 			>
-				<label htmlFor="selectTrigger">
-					{Liferay.Language.get('retrieve-type')}
+				<label>
```
Comment (ambrinchaudhary) [inline modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx:256]: question: why did you remove the htmlFor attribute? it makes sense for labels

File: modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx [hunk 2]
```diff
@@ -180,82 +248,105 @@ export default function BaseAPIEndpointFields({
...
 			</ClayForm.Group>
 
+			{selectedHttpMethod?.value === 'get' && (
+				<ClayForm.Group
+					className={classNames({
+						'has-error': displayError.retrieveType,
+					})}
+				>
+					<label htmlFor="selectTrigger">
+						{Liferay.Language.get('Retrieve type')}
```
Comment (ambrinchaudhary) [inline modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx:298]: use a key `retrieve-type`

File: modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx [hunk 3]
```diff
@@ -355,56 +446,57 @@ export default function BaseAPIEndpointFields({
...
-											? removeLeadingForwardSlash(
-													data.parameter
-											  )
-											: ''
-									}
-								/>
-							</ClayInput.GroupItem>
-						</>
-					)}
+					{selectedHttpMethod?.value === 'get' &&
```
Comment (ambrinchaudhary) [inline modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx:449]: these `'get'`,` 'retrieveType'`, `'single-element'`... that I see along the code should be constants.

File: modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx [hunk 4]
```diff
@@ -148,6 +179,43 @@ export default function BaseAPIEndpointFields({
 		}
 	}, [data, scopeOptions]);
 
+	useEffect(() => {
+		if (data.httpMethod?.key && httpMethodOptions.length) {
```
Comment (ambrinchaudhary) [inline modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx:183]: Here there are things that I don't understand, this useEffect looks "difficult" and that could end in bugs. Let's talk tomorrow to see how can we improve this.

**Review by ambrinchaudhary — COMMENTED** [review]
Just a quick review, I will need to see this working to better understand what it is done ;-)

File: modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx (hunk 1 above)
Comment (jaimelr10) [reply modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx:256]: I'll add it!

File: modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx (hunk 4 above)
Comment (ambrinchaudhary) [reply modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx:183]: Already spoke about this. We will leave it as it is, and will change and divide in subcomponents in the future.

**Comment by jaimelr10** [comment]
ci:test:sf

**Comment by jaimelr10** [comment]
ci:forward

**Comment by jaimelr10** [comment]
ci:forward:force

**Comment by jaimelr10** [comment]
ci:forward:force

**Comment by jaimelr10** [comment]
ci:forward:force

**Comment by jaimelr10** [comment]
ci:forward:force
